docs/design/numeric_literals.md

```diff
@@ -0,0 +1,241 @@
+
+// This initializes y to 2^60.
+var y: i64 = 1 << 60;
+
+// This forms a rational literal whose value is one third, and converts it to
+// the nearest representable value of type `f64`.
+var z: f64 = 1.0 / 3.0;
+
+// This is an error: 300 cannot be represented in type `i8`.
+var c: i8 = 300;
+
+fn f[template T:! Type](v: T) {
+  var x: i32 = v * 2;
+}
+
+// OK: x = 2_000_000_000.
+f(1_000_000_000);
+
+// Error: 4_000_000_000 can't be represented in type `i32`.
+f(2_000_000_000);
+
+// No storage required for the bound when it's of integer literal type.
+struct Span(template T:! Type, template BoundT:! Type) {
+  var begin: T*;
+  var bound: BoundT;
+}
+
+// Returns 1, because 1.3 can implicitly convert to f32, even though conversion
+// to f64 might be a more exact match.
+fn G() -> i32 {
+  match (1.3) {
+    case _: f32 => { return 1; }
+    case _: f64 => { return 2; }
+  }
+}
+
+// Can only be called with a literal 0.
+fn PassMeZero(_: IntLiteral(0));
+
+// Can only be called with integer literals in the given range.
+fn ConvertToByte[template N:! BigInt](_: IntLiteral(N)) -> i8
+    if N >= -128 and N <= 127 {
+  return N as i8;
+}
+
+// Given any int literal, produces a literal whose value is one higher.
+fn OneHigher(L: IntLiteral(template _:! BigInt)) -> auto {
+  return L + 1;
+}
+// Error: 256 can't be represented in type `i8`.
+var v: i8 = OneHigher(255);
+```
+
+## Numeric literal semantics
+
+Numeric Literal semantics are covered in the
+[numeric_literals](lexical_conventions/numeric_literals.md) lexical conventions
+doc. Decimal and Real-Number semantics are defined, with decimal, hexadecimal
+and binary integer literals, and decimal and hexadecimal real number literals.
+
+## Open questions
+
+### Primitive types as code vs built-in
+
+There are open questions about the extent to which these types should be defined
+in Carbon code rather than special. Clearly they can't be directly implemented
+w/o help, but it might still be useful to force the programmer-observed
+interface to reside in code. However, this can cause difficulty with avoiding
+the need to import things gratuitously.
+
+### Syntax for wrapping operations
+
+Open question around allowing special syntax for wrapping operations (even on
+signed types) and/or requiring such syntax for wrapping operations on unsigned
+types.
+
+### Non-power-of-two sizes
+
+Supporting non-power-of-two sizes is likely needed to have a clean model for
+bitfields, but requires more details to be worked out around memory access.
+
+## References
+
+> -   Proposal
+>     [#144: Numeric literal semantics](https://github.com/carbon-language/carbon-lang/pull/144)
+> -   Proposal
+>     [#143: Numeric literals](https://github.com/carbon-language/carbon-lang/pull/143)
+> -   Proposal
+>     [#820: Implicit conversions](https://github.com/carbon-language/carbon-lang/pull/820)
+> -   Proposal
+>     [#2015: Numeric type literal syntax](https://github.com/carbon-language/carbon-lang/pull/2015)
```

Comment:
How are these proposals referenced in this document? I suspect they should be omitted.
```suggestion
```